drivers/api/no_os_spi.c

```diff
@@ -58,33 +65,94 @@ int32_t no_os_spi_init(struct no_os_spi_desc **desc,
 
 	if (!param->platform_ops->init)
 		return -ENOSYS;
-
+	if (param->device_id > SPI_MAX_BUS_NUMBER)
+		return -EINVAL;
+	//Initializing BUS descriptor
```

Comment:
This PR looks good to me now.
1 comment though about aesthetics, you seem to have a lot of these `//Text` comments whereas we always leave a blank after `// Text`, please insert those whitespaces, otherwise no-OS code will look very heterogenous :D